src/equisolve/numpy/scripts/equi_kit.py

```diff
@@ -0,0 +1,299 @@
+        #
+        #  join(X0_t, ..., XN_t)
+        #           X_jt
+        #
+        #  estimator.fit(X_jt, y_t)
+
+        script.forward(Xi)
+    ```
+
+    COMMENT class name is too general, can we distinguish it to Hamiltonian learning and CG-NN applications where postprocessing happens after the join to compute errors?
+            Might be not important because these applications require gradient optimization, so they would be in any not within the Scikit scope
+
+    COMMENT The logic is a bit unnecessary entangled with atomistic data, but I would not bother with it for now
+    """
+    def __init__(self, hypers, *, feature_aggregation="mean", transformer_X=None, transformer_y=None, estimator=None):
+        self.hypers = hypers
+
+        self.feature_aggregation = feature_aggregation
+
+        if estimator is None:
+            raise NotImplemented("Ridge still needs a good default alpha value, before we can support default parameter")
+            #empty_tm = TensorMap(keys=Labels.single(), blocks=[TensorBlock(
+            #    values=np.empty((0,1)),
+            #    samples=Labels(names=["samples"], values=np.empty((0,1), dtype=np.int32)),
+            #    components=[],
+            #    properties=Labels.single(),
+            #)])
+            #self.estimator = Ridge(parameter_keys=parameter_keys, alpha=empty_tm)
+
+    def fit(self, X: Tuple[TensorMap, ...], y: TensorMap, **kwargs) -> TEquiKitScript:
+        # X : (X0, X1, ..., XN)
+        self._set_and_check_fitting_parameters()
+        if "transformer_X" not in kwargs.keys():
+            kwargs["transformer_X"] = {}
+        if "transformer_y" not in kwargs.keys():
+            kwargs["transformer_y"] = {}
+        if "estimator" not in kwargs.keys():
+            kwargs["estimator"] = {}
+
+        if self.transformer_X is None:
+            self._transformers_X = None
+        else:
+            self._transformers_X = []
+            for i in range(len(T)):
+                self._transformers_X.append( deepcopy(transformer_X).fit(X[i], **kwargs["transformer_X"]) )
+
+        if self.transformer_y is None:
+            self._transformer_y = None
+        else:
+            self._transformer_y = deepcopy(self.transformer_y).fit(y, **kwargs["transformer_y"])
+
+        X = tuple(self._transformers_X[i].transform(X[i]) for i in range(len(self._transformers_X)))
+        y = self._transformer_y.transform(y)
+
+        X = self.transform_join(X)
+
+        self._estimator = deepcopy(self.estimator).fit(X, y, **kwargs["estimator"])
+
+    def forward(self, X: Tuple[TensorMap, ...]) -> TensorMap:
+        # TODO check if is fitted
+
+        if self._transformers_X is not None:
+            X = tuple(self._transformers_X[i].transform(X[i]) for i in range(len(self._transformers_X)))
+        if self._transformer_y is not None:
+            y = self._transformer_y.transform(y)
+
+        X = self.join(X)
```

Comment:
I think join should only called here something like

if len(X) > 1. With this 
```suggestion
        if len(X) > 1:
            try:
                X = self.join(X)
            except NotImplemented as e:
                raise NotImplemented('More than one X. But join function is not implemented!') from e
```